timApp/tests/server/test_velp.py

```diff
@@ -295,3 +303,83 @@ def check_ann_for_testuser1(ann: dict):
         info_anns = self.get("/settings/info")["annotations"]
         for iann in info_anns:
             check_ann_for_testuser1(iann)
+
+
+class VelpGroupDeletionTest(TimRouteTest):
+    def test_delete_velp_group(self):
+        # set up docs and velp groups
+        self.login_test1()
+        d = self.create_doc(title="test velp group delete")
+        self.test_user_2.grant_access(d, AccessType.view)
+
+        g = self.json_post(
+            f"/{d.document.id}/create_velp_group",
+            {"name": "test-group1", "target_type": 1},
+        )
+        db.session.commit()
+
+        # Case 1:
+        # Test user 2 should not be able to delete velp group with view permissions
+        self.login_test2()
+        # try removing from directory
+        self.delete(url=f"/documents/{g['id']}", expect_status=403)
+        # try deleting from database
+        self.delete(
+            url=f"/delete_velp_group_from_database/{g['id']}", expect_status=403
+        )
+
+        # Case 2:
+        # Should not be able to delete with edit permissions
+        self.test_user_2.grant_access(d, AccessType.edit)
```

Comment:
`grant_access()` antaa oikeuden, mutta ei vielä lähetä muutokset kantaan (siitä puuttuu tarvittava kutsu, jotta samassa sessiossa voi antaa oikeudet monelle). Kyseisen funktion jälkeen pitää vielä kutsua
`db.session.commit()`. Muuten se nyt testaa tapaukset 1-4 view-oikeudella (eli tapaus 2, 3, 4 testaavat tapauksen 1).

Reply:
Ok, testasin kyllä molemmin päin ja meni läpi, joten poistin `db.session.commit()`it turhina, mutta taitaa tosiaan olla että mulla on muualla logiikkavirhe tarkistuksissa.

Reply:
Joo, se nyt siis pitää testuser2:n oikeudet `view`-tasolla, minkä takia tapaukset 1-4 menevät läpi. Ongelman olisi voinut ehkä huomata, jos tapauksen 5 tekisi myös testuser2:lla (eli yrttäisi antaa sille owner-oikeuden).